applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c

```diff
@@ -313,6 +346,20 @@ static void client_init(void)
 	client.tx_buf = tx_buffer;
 	client.tx_buf_size = sizeof(tx_buffer);
 
+	/* MQTT Will configuration */
+	strcpy((char *)pub_topic, mqtt_clientid);
```

Comment:
I think that this overwrites unwanted memory. pub_topic is not a memory area. It works by having a pointer to memory area.

Reply:
`static uint8_t pub_topic[MQTT_MAX_TOPIC_LEN];`
It's not used until MQTTPUB.

Reply:
This was my bad, somehow got mqtt_topic and pub_topic confused.